workflows/src/main/kotlin/hk/edu/polyu/af/bc/badge/flows/assertion/RevokeAssertionFlow.kt

```diff
@@ -1,10 +1,41 @@
 package hk.edu.polyu.af.bc.badge.flows.assertion
 
+
+import com.r3.corda.lib.tokens.contracts.NonFungibleTokenContract
+import com.r3.corda.lib.tokens.contracts.states.EvolvableTokenType
+import com.r3.corda.lib.tokens.contracts.states.NonFungibleToken
+import com.r3.corda.lib.tokens.contracts.types.TokenPointer
+import com.r3.corda.lib.tokens.contracts.types.TokenType
 import hk.edu.polyu.af.bc.badge.states.Assertion
+import hk.edu.polyu.af.bc.badge.states.BadgeClass
+import jdk.nashorn.internal.ir.annotations.Immutable
+import net.corda.core.contracts.*
+import net.corda.core.flows.CollectSignatureFlow
+import net.corda.core.flows.FinalityFlow
 import net.corda.core.flows.FlowLogic
 
-class RevokeAssertionFlow (val assertion: Assertion): FlowLogic<Unit>() {
-    override fun call(): Unit {
-        assertion.revoked=false
+import net.corda.core.node.services.Vault
+import net.corda.core.node.services.queryBy
+import net.corda.core.node.services.vault.QueryCriteria
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+import java.util.*
+
+
+class RevokeAssertionFlow (
```

Comment:
The naming convention of flows should follow the practice of the account-sdk. The general rule is that if your flow initiate sessions, it should not have `Flow` appended.

Reply:
And I don't think you have updated the state. I thought we agree that by revoking an `Assertion`, its `revoked` turns to `true`. Here you just eliminate the state.